coverage.cmake

```diff
@@ -0,0 +1,45 @@
+#
+#   Copyright 2022 CNRS
+#
+#   Author: Guilhem Saurel
+#
+#
+
+#.rst:
+# .. ifmode:: internal
+#
+#   .. variable:: ENABLE_COVERAGE
+#
+#      When this is ON, coverage compiler flags are enabled
+
+option(ENABLE_COVERAGE "enable C++ and python code coverage" OFF)
```

Comment:
I think the option shouldn't exist if the compiler doesn't support the `--coverage` flag, the following is probably good enough:

```cmake
if(NOT MSVC)
  option(ENABLE_COVERAGE ...)
else()
  set(ENABLE_COVERAGE OFF)
endif()
```